enhancements/microshift/microshift-telemetry.md

```diff
@@ -0,0 +1,299 @@
+* Number of container images on disk
+* Number of routes/ingress/services
+* Number of CRDs
+* OS version / Type (rpm / ostree)
+
+These metrics should be sent at least once a day.
+
+### Metrics details
+Based on [Telemetry API](https://rhobs-handbook.netlify.app/products/openshiftmonitoring/telemetry.md/)
+MicroShift will send metrics using Prometheus format. There are several ways
+of using the API: [using the client to forward metrics](https://github.com/openshift/telemeter/tree/main?tab=readme-ov-file#upload-endpoint-receive-metrics-in-client_modelmetricfamily-format-from-telemeter-client-currently-used-by-cmo),
+or [direct write requests](https://github.com/openshift/telemeter/tree/main?tab=readme-ov-file#metricsv1receive-endpoint-receive-metrics-in-prompbwriterequest-format-from-any-client).
+Using the client is not an option because that requires a local Prometheus
+instance, not viable in the typical resource constrained MicroShift deployment.
+The direct write requests take raw Prometheus data and require both special
+labels (`_id` for the cluster id) and crafting the authentication headers
+(cluster id and pull secret in a specific format in HTTP headers).
+
+OpenShift is using the same API and the same backend, so we need a way to
+distinguish MicroShift in the pool of metrics. For this we can use labels, as
+it does not require applying for new supported metrics. MicroShift metrics
+labels summary:
+| Label | Values | Approximate max size |
+|:---|:---|:---|
+|_id|cluster id|40B|
+|label_kubernetes_io_arch|amd64, aarch64|32B|
+|resource|Used to specify K8s resource names: pods, namespaces, etc.|20-50B|
+|version|microshift version|16B|
+|ostree_commit|OStree commit id, if the system is deployed using ostree. This helps identifying fleets|80B|
+
+Metrics from MicroShift are already supported in the API because OpenShift is
+using them. List follows:
+* cluster:capacity_cpu_cores:sum. Number of allocated CPUs for MicroShift.
+* cluster:capacity_memory_bytes:sum. Number of bytes of memory allocated for
+  MicroShift.
+* cluster:cpu_usage_cores:sum. Usage of CPU in percentage.
+* cluster:memory_usage_bytes:sum. Usage of memory in percentage.
+* cluster:usage:resources:sum. Usage of k8s resources, in count. Number of
+  pods, namespaces, services, etc.
+* cluster:usage:containers:sum. Number of active containers.
+* cluster_version. Information about installed version.
+* instance:etcd_object_counts:sum. Number of objects etcd contains.
+
+Combining metrics with their labels:
+| Metric | Labels |
+|:---|:---|
+|cluster:capacity_cpu_cores:sum|_id, label_kubernetes_io_arch, ostree_commit|
+|cluster:capacity_memory_bytes:sum|_id, label_kubernetes_io_arch, ostree_commit|
+|cluster:cpu_usage_cores:sum|_id, ostree_commit|
+|cluster:memory_usage_bytes:sum|_id, ostree_commit|
+|cluster:usage:resources:sum|_id, ostree_commit, resource|
+|cluster:usage:containers:sum|_id, ostree_commit|
+|cluster_version|_id, ostree_commit, version|
```

Comment:
I'd advocate for a different metric name otherwise it might break other users of the existing `cluster_version` since the labels are going to be different.

Reply:
This makes sense. Would using different values for the `type` label help in any way? We would like to avoid creating dependencies on other teams, if possible.

Reply:
> Would using different values for the type label help in any way?

If 2 different systems emit the same metric, we want ideally that the metrics have the same semantics and that the label sets are the same. Otherwise it makes querying less robust (queries might assume that a particular label is always present). In particular `cluster_version` is a base metric used to derive other Telemetry metrics.

Reply:
Agree with keeping semantics. Will apply for a new metric because this one serves as a base to many other things and we do not want to disrupt OpenShift metrics queries.